### ISSUE TITLE
insert() asserts when host portion non-zero

### ISSUE THREAD

**Issue by pusateri** [hroi/treebitmap#9]
Hitting this assert when adding a interface address/prefix length into a table. I didn't zero out the host portion of the address before trying to insert it and it crashed.

I think it would be better if insert() returned None in this case. Or maybe insert() should return a Result with a value or an error?

thread 'insert_assert' panicked at 'assertion failed: ret > 0', src/tree_bitmap/node.rs:66:5

This test reproduces the assert:
```
#[test]
fn insert_assert() {
    let mut tbm = IpLookupTable::new();
    let ip1 = Ipv4Addr::from_str("10.255.71.185").unwrap();
    tbm.insert(ip1, 22, 1);
```

**Comment by hroi** [comment]
It's a programmer error to insert invalid prefixes, I should add a note about panicing on invalid input. I don't think it's worth complicating the return type to handle invalid input.

**Comment by hroi** [comment]
Fixed in 56c6f4717ebbddca010d7799af810a6f39ad5ec9

**Comment by pusateri** [comment]
I don't think it's a programmer error to want to insert interface addresses with prefix lengths into a tree. While the user can mask out the host portion, why should they have to? The insert code should just do this automatically in all cases.
